### PR TITLE
Update figshare deserialization of submitted values, and update tests

### PR DESCRIPTION
**Pull request by samchrisinger**
## Purpose

Changes to figshare folder serialization/deserialization broke figshare_config_put, and corresponding tests. This fixes those issues.

## Changes

- Update folder serialization/deserialization
- Update test to reflect this

## Side Effects

None

### CODE REVIEW

File: website/addons/figshare/views/config.py [hunk 1]
```diff
@@ -38,16 +38,23 @@ def figshare_config_put(node_addon, auth, **kwargs):
     fields = request.json.get('selected', {})
     node = node_addon.owner
 
-    folder = {
-        'name': fields.get('name') or '',
-        'id': fields.get('id') or None,
-        'type': fields.get('type') or None,
-    }
-    node_addon.update_fields(folder, node, auth)
+    name = fields.get('name') or ''
```
Comment (sloria) [inline website/addons/figshare/views/config.py]: In the future, we should probably make the title/name semantics consistent across request arguments, the DB schema, and responses, but this is fine for now.

Comment (samchrisinger) [inline website/addons/figshare/views/config.py]: Agreed, was going for minimum change to the code. 

I'm happy to change this if you'd prefer.

File: website/addons/figshare/views/config.py [hunk 2]
```diff
@@ -38,16 +38,23 @@ def figshare_config_put(node_addon, auth, **kwargs):
...
+    fs_id = fields.get('id') or None
```
Comment (sloria) [inline website/addons/figshare/views/config.py]: The `or None` is unnecessary.

File: website/addons/figshare/views/config.py (hunk 1 above)
Comment (sloria) [inline website/addons/figshare/views/config.py]: If we always pass this value to `update_fields`, won't that result in the node settings' figshare title to be an empty string if the client doesn't pass `name`?

Comment (sloria) [inline website/addons/figshare/views/config.py]: @samchrisinger No need to make the "title"/"name" in this PR

**Comment by sloria** [comment]
Pass finished. :jack_o_lantern:

File: website/addons/figshare/views/config.py (hunk 2 above)
Comment (sloria) [inline website/addons/figshare/views/config.py]: Also, to get nitpicky... let's avoid acronyms in variable names. `figshare_id`, `figshare_type` would be an improvement.

File: website/addons/figshare/views/config.py (hunk 1 above)
Comment (sloria) [inline website/addons/figshare/views/config.py]: Related to my previous comment re: `update_fields`: won't `id` and `type` also be set to `None` if the client doesn't provide them?

I'm not sure this is the desired behavior.

Comment (samchrisinger) [inline website/addons/figshare/views/config.py]: I'm going to catch these with a 400 now

Comment (sloria) [inline website/addons/figshare/views/config.py]: :+1:

File: website/addons/figshare/views/config.py [hunk 3]
```diff
@@ -38,16 +38,31 @@ def figshare_config_put(node_addon, auth, **kwargs):
...
-        'id': fields.get('id') or None,
-        'type': fields.get('type') or None,
-    }
-    node_addon.update_fields(folder, node, auth)
+    name = fields.get('name')
+    figshare_id = fields.get('id')
+    figshare_type = fields.get('type')
+
+    if not name or not figshare_id or not figshare_type:
+        return HTTPError(http.BAD_REQUEST, message="""
```
Comment (sloria) [inline website/addons/figshare/views/config.py]: > Use parentheses for line continuations.

https://cosdev.readthedocs.org/en/latest/style_guides/python.html#line-lengths

Also, I'm not sure how these line breaks and indents will display in a JSON response. Maybe we go with a shorter message, like: "Must provide name, id, and type".

Comment (sloria) [inline website/addons/figshare/views/config.py]: You must raise the HTTPError, not return it.

Also, in order for the message to be included in the response body, you need to do:

```python
raise HTTPError(http.BAD_REQUEST, data=dict(
    message_long='You must provide name, id, and type'
))
```

File: website/addons/figshare/views/config.py [hunk 4]
```diff
@@ -38,16 +38,31 @@ def figshare_config_put(node_addon, auth, **kwargs):
...
-        'name': fields.get('name') or '',
-        'id': fields.get('id') or None,
-        'type': fields.get('type') or None,
-    }
-    node_addon.update_fields(folder, node, auth)
+    name = fields.get('name')
+    figshare_id = fields.get('id')
+    figshare_type = fields.get('type')
+
+    if not name or not figshare_id or not figshare_type:
```
Comment (sloria) [inline website/addons/figshare/views/config.py]: Might be more concise and readable to do:

```
if not all([name, figshare_id, figshare_type]):
   # ...

**Comment by sloria** [comment]
Pass finished. :ear:

**Pull request merged**